lib/vue-segment/vue-segment.js

```diff
@@ -1,4 +1,59 @@
 export default function (Vue, options) {
+  /**
+   * Segment Source middleware to modify each Segment payload
+   */
+  function payloadComplianceMiddleware ({ payload, next, integrations }) {
+    // Keys to remove from each Segment payload
+    const payloadKeysToRemove = {
+      properties: ['url', 'path', 'title', 'search', 'referrer']
+    }
+
+    // Remove keys from the segment payload to prevent property values from being overwritten by Segment
+    const deleteTargetPayloadKeys = (payload, keysToRemove, path = []) => {
+      const payloadCopy = { ...payload }
+
+      Object.keys(keysToRemove)
+        .forEach((currentKey) => {
+          const currentProp = keysToRemove[currentKey]
+          path = [...path, currentKey]
+
+          if (Array.isArray(currentProp)) {
+            const targetObj = path.reduce((prev, curr) => prev?.[curr], payloadCopy)
+
+            currentProp.forEach((prop) => {
+              delete targetObj[prop]
+            })
+          } else {
+            deleteTargetPayloadKeys(payloadCopy, currentProp, path)
```

Comment:
does this recursively delete a given key in payloadKeysToRemove? Not sure if we need/want that if so.

Reply:
Yeah, this is recursive. Wasn't sure if we were going to have to add to the keys we are removing at some point. But if this is all we are going to need for now and future, I can adjust this function